app/candidates/components/map/Map.js

```diff
@@ -224,24 +225,15 @@ const Map = () => {
 
   // Listen for URL changes to update the map when back button is pressed
   useEffect(() => {
-    const handleRouteChange = (url) => {
-      const urlParams = new URLSearchParams(url.split('?')[1]);
-      const lat = parseFloat(urlParams.get('mapCenterLat'));
-      const lng = parseFloat(urlParams.get('mapCenterLng'));
-      const zoomLevel = parseInt(urlParams.get('zoom'), 10);
-
-      if (mapRef.current && !isNaN(lat) && !isNaN(lng) && !isNaN(zoomLevel)) {
-        mapRef.current.setCenter({ lat, lng });
-        mapRef.current.setZoom(zoomLevel);
-      }
-    };
-
-    router.events.on('routeChangeComplete', handleRouteChange);
+    const lat = parseFloat(searchParams.get('mapCenterLat'));
+    const lng = parseFloat(searchParams.get('mapCenterLng'));
+    const zoomLevel = parseInt(searchParams.get('zoom'), 10);
 
-    return () => {
-      router.events.off('routeChangeComplete', handleRouteChange);
-    };
-  }, [router]);
+    if (mapRef.current && !isNaN(lat) && !isNaN(lng) && !isNaN(zoomLevel)) {
+      mapRef.current.setCenter({ lat, lng });
+      mapRef.current.setZoom(zoomLevel);
+    }
+  }, [searchParams.toString()]);
```

Comment:
Won't this create a new string object reference in memory on each render, leading to this running every time regardless?

Reply:
Hmm.. looks like I'm wrong.. 🤷 

<img width="470" alt="image" src="https://github.com/user-attachments/assets/71de30a3-b1c9-4001-96c6-b18e24c8e407">